File: src/_02_StructuralPattern/_02_05_Facade/java/after/EmailSenderFacade.java
package _02_StructuralPattern._02_05_Facade.java.after;

import java.util.Properties;

import _02_StructuralPattern._02_05_Facade.java.after.EmailLib.InternetAddress;
import _02_StructuralPattern._02_05_Facade.java.after.EmailLib.MimeMessage;
import _02_StructuralPattern._02_05_Facade.java.after.EmailLib.Session;
import _02_StructuralPattern._02_05_Facade.java.after.EmailLib.Transport;

public class EmailSenderFacade {

  private EmailSettingsFacade emailSettings;

  public EmailSenderFacade(EmailSettingsFacade emailSettings) {
    this.emailSettings = emailSettings;
  }

  public void sendEmail(EmailMessageFacade emailMessage) {
    Properties properties = new Properties();
    properties.setProperty("mail.smtp.host", emailSettings.getHost());

    Session session = Session.getDefaultInstance(properties);

    try {
      MimeMessage message = new MimeMessage(session);
      message.setFrom(new InternetAddress(emailMessage.getFrom()));
      message.addRecipient(new InternetAddress(emailMessage.getTo()));
      message.setSubject(emailMessage.getSubject());
      message.setText(emailMessage.getText());

      Transport.send(message);
    } catch(Exception e) {
      e.printStackTrace();
    }
  }
  
}

File: src/_02_StructuralPattern/_02_05_Facade/EmailSenderFacade.js
import Session from "./EmailLib/Session.js";
import InternetAddress from "./EmailLib/InternetAddress.js";
import MimeMessage from "./EmailLib/MimeMessage.js";
import Transport from "./EmailLib/Transport.js";

/**
 * @typedef { import("./EmailSettingsFacade").default } EmailSettingsFacade
 * @typedef { import("./EmailMessageFacade").default } EmailMessageFacade
 */

export default class EmailSenderFacade {
  /** @type { EmailSettingsFacade } */
  #emailSettings;

  /** @param { EmailSettingsFacade } emailSettings */
  constructor(emailSettings) {
    this.#emailSettings = emailSettings;
  }

  /** @param { EmailMessageFacade } emailMessage */
  sendEmail(emailMessage) {
    const properties = new Map();
    properties.set("mail.smtp.host", this.#emailSettings.getHost());

    const session = new Session(properties);

    const message = new MimeMessage(session);
    message.setFrom(new InternetAddress(emailMessage.getFrom()));
    message.addRecipient(new InternetAddress(emailMessage.getTo()));
    message.setSubject(emailMessage.getSubject());
    message.setText(emailMessage.getText());

    Transport.send(message);
  }
}